### PR TITLE
help cmd as per #141

### PR DESCRIPTION
**Pull request by vaibhavgarg237**

### CODE REVIEW

**Comment by darosior** [comment]
Can you remove 5441068 please?

File: src/daemon/jsonrpc/api.rs [hunk 1]
```diff
@@ -333,6 +337,123 @@ impl RpcApi for RpcImpl {
         }))
     }
 
+    fn help(&self, meta: Self::Metadata) -> jsonrpc_core::Result<serde_json::Value> {
+        Ok(json!(
+        {
+            "Commands":[
```
Comment (darosior) [inline src/daemon/jsonrpc/api.rs:348]: lowercase in fields please

**Review by darosior — COMMENTED** [review]
Looks good from a quick skim, but you didn't `cargo fmt`. I wonder if we could have something more generic (eventually?).

**Comment by vaibhavgarg237** [comment]
> Looks good from a quick skim, but you didn't `cargo fmt`. I wonder if we could have something more generic (eventually?).

Can you please elaborate a little more about making it more generic? ( Something like importing commands from a separate file? )

**Comment by darosior** [comment]
I was more musing about the future, this is the right approach for now!
‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

File: src/daemon/jsonrpc/api.rs [hunk 2]
```diff
@@ -91,6 +91,10 @@ pub trait RpcApi {
     #[rpc(meta, name = "getinfo")]
     fn getinfo(&self, meta: Self::Metadata) -> jsonrpc_core::Result<serde_json::Value>;
 
+    // Print all available commands
```
Comment (darosior) [inline src/daemon/jsonrpc/api.rs:94]: nit: This should be a [doc comment](https://doc.rust-lang.org/reference/comments.html) to match all other methods

File: src/daemon/jsonrpc/api.rs [hunk 3]
```diff
@@ -333,6 +337,123 @@ impl RpcApi for RpcImpl {
...
 
+    fn help(&self, meta: Self::Metadata) -> jsonrpc_core::Result<serde_json::Value> {
+        Ok(json!(
+        {
+            "commands":[
+                {
+                  "name": "getinfo",
+                  "parameters": [
+
+                  ],
```
Comment (darosior) [inline src/daemon/jsonrpc/api.rs:348]: For all empty arrays of parameters, can you put the square brackets on the same line? ie:
```json
"parameters": [],
```

File: src/daemon/jsonrpc/api.rs [hunk 4]
```diff
@@ -333,6 +337,123 @@ impl RpcApi for RpcImpl {
         }))
     }
 
+    fn help(&self, meta: Self::Metadata) -> jsonrpc_core::Result<serde_json::Value> {
```
Comment (darosior) [inline src/daemon/jsonrpc/api.rs:340]: You don't use `meta` here, yet you have to keep it in the function declaration. To indicate you don't use it (and quiet the compiler warning), please prefix the variable name with an underscore (or just use an underscore instead):
```diff
diff --git a/src/daemon/jsonrpc/api.rs b/src/daemon/jsonrpc/api.rs
index 76a4c79..e11a11d 100644
--- a/src/daemon/jsonrpc/api.rs
+++ b/src/daemon/jsonrpc/api.rs
@@ -337,7 +337,7 @@ impl RpcApi for RpcImpl {
         }))
     }
 
-    fn help(&self, meta: Self::Metadata) -> jsonrpc_core::Result<serde_json::Value> {
+    fn help(&self, _: Self::Metadata) -> jsonrpc_core::Result<serde_json::Value> {
         Ok(json!(
         {
             "commands":[
```

File: src/daemon/jsonrpc/api.rs [hunk 5]
```diff
@@ -333,6 +337,123 @@ impl RpcApi for RpcImpl {
...
+                {
+                    "name": "getrevocationtxs",
+                    "parameters": [
+                        "outpoint"
+                    ],
+                    "description": "Retrieve the Revault revocation transactions to sign",
+                },
+                {
+                    "name":"getunvaulttx",
+                    "parameters":[
```
Comment (darosior) [inline src/daemon/jsonrpc/api.rs:360]: nit: missing a space between `:` and `[`

Comment (darosior) [reply src/daemon/jsonrpc/api.rs:360]: Same for every object below

**Review by darosior — COMMENTED** [review]
Some more nits

**Comment by vaibhavgarg237** [comment]
> Some more nits

made required changes & clubbed in single commit...

**Comment by darosior** [comment]
Looks good to me, utACK 9238ecb41752e097443d0bc117df35ebd4ac932e

**Comment by danielabrozzoni** [comment]
I noticed just now that it's a bit hard to navigate - that's my fault, I didn't notice when I wrote the spec for the command. For the future, maybe we should switch to having the name of the command as the key, like:
```
[
  {
    "getinfo": { ... }
  },
  {
    "getrevocationtxs": { ... }
  }
]
```

But for now it's good as it is! Thanks :)

ACK acbc03f436ca112bc8addefbc1882c80064c118e

File: src/daemon/jsonrpc/api.rs [hunk 6]
```diff
@@ -333,6 +337,105 @@ impl RpcApi for RpcImpl {
...
     }
 
+    fn help(&self, _: Self::Metadata) -> jsonrpc_core::Result<serde_json::Value> {
+        Ok(json!(
+        {
+            "commands": [
+                {
+                  "name": "getinfo",
+                  "parameters": [],
+                  "description": "Display general information",
```
Comment (darosior) [inline src/daemon/jsonrpc/api.rs:347]: Just noticed when merging that indentation is off..

**Review by darosior — COMMENTED** [review]
Also, we should probably mention it in `doc/API.md`

**Pull request merged**